tests/src/test-agent.ts

```diff
@@ -187,15 +187,66 @@ test("stats are acquired", async (t) => {
   expectResponse(t, nats, "drone.1.spawn", request, true)
   await sleep(100)
 
-  const statsStatusSubsription = 
+  const statsStatusSubsription =
     new NatsMessageIterator<unknown>(
-      nats.subscribe(`backend.${backendId}.stats`, { timeout : 1000000 })
-  )
+      nats.subscribe(`backend.${backendId}.stats`, { timeout: 1000000 })
+    )
   const statsMessage = await statsStatusSubsription.next()
-  const stats= statsMessage[0]
+  const stats = statsMessage[0]
   t.assert(stats["cpu_use_percent"] > 0 && stats["mem_use_percent"] > 0)
 })
 
+test("stats are killed after container dies", async (t) => {
+  const backendId = generateId()
+  const natsPort = await t.context.docker.runNats()
+  await sleep(1000)
+  const nats = await connect({ port: natsPort, token: "mytoken" })
+  t.context.runner.runAgent(natsPort)
+
+  await expectMessage(t, nats, "drone.register", {
+    cluster: "mydomain.test",
+    ip: "123.12.1.123"
+  }, {
+    Success: {
+      drone_id: 1,
+    }
+  })
+
+  await sleep(100)
+
+  const request: SpawnRequest = {
+    image: TEST_IMAGE,
+    backend_id: backendId,
+    max_idle_secs: 10,
+    env: {
+      PORT: "8080",
+    },
+    metadata: {},
+  }
+
+  expectResponse(t, nats, "drone.1.spawn", request, true)
+
+  await sleep(100)
+
+  const statsStatusSubsription =
+    new NatsMessageIterator<unknown>(
+      nats.subscribe(`backend.${backendId}.stats`, { timeout: 1000000 })
+    )
+
+  try {
+    while (true) {
+      //NOTE: this sleep is strongly coupled to 
+      //DEFAULT_DOCKER_THROTTLED_STATS_INTERVAL_SEC in agent/docker.rs
+      await Promise.race([statsStatusSubsription.next(), sleep(11000)])
```

Comment:
I don't like using exceptions for happy-path control flow.

I think this should work?

```
let success = await Promise.race([statsStatusSubsription.next(), sleep(11000)])
if (!success) {
  break;
}
```

And then remove the outer `try / catch`